### ISSUE TITLE
Create GiST Index on `properties` in Persons table for faster `api/person?search`

### ISSUE THREAD

**Issue by fuziontech** [PostHog/posthog#10648]
## In what situation are you experiencing subpar performance?

If you visit https://app.posthog.com/persons and search for a person via email or distinct_id you'll notice that it can be very slow and commonly times out.

## Current impact

A few customers have reported that person search is unusable.

## Why is this slow?

These queries are going to Postgres
https://github.com/PostHog/posthog/blob/master/posthog/api/person.py#L134

```python
   def search_filter(self, queryset, attr, value, *args, **kwargs):
        return queryset.filter(
            Q(properties__icontains=value) | Q(persondistinctid__distinct_id__icontains=value)
        ).distinct("id")
```
When this endpoint queries Postgres it's running an `ILIKE` on the `properties` field of `posthog_person` and the `distinct_id` field on `person_persondistinctid`
https://docs.djangoproject.com/en/4.0/ref/models/querysets/

Neither of these fields have any search optimized indexes (like [GIN or GiST](https://www.compose.com/articles/indexing-for-full-text-search-in-postgresql/)).
https://app.pganalyze.com/databases/-675880137/tables/53506540143/indexes

This means that every call Postgres is going to full text search every person properties and distinct_id that belongs to the team conducting the search. The person table is getting quite large these days and the teams using this search feature are also getting quite large.

## What should we do?

- We move the search process to ClickHouse (ideal, but will require some refactoring)
- Until then, we should add a GiST index to `properties` on `posthog_person` table for faster `ILIKE` queries

**Comment by neilkakkar** [comment]
Linking existing issues (and closing old ones because this has more actionables):

#8660 #8391

**Comment by jamesefhawkins** [comment]
sidenote: is this the reason we also don't search persons here?

<img width="490" alt="Screenshot 2022-07-07 at 11 48 02" src="https://user-images.githubusercontent.com/47497682/177756244-fe6e50e4-6140-46e2-a1d0-1db9f9a634be.png">

**Comment by neilkakkar** [comment]
Exactly, I removed it from overall search because it would time out too often, leading to a worse experience. As soon as this issue is fixed, we can bring it back to universal search as well

**Comment by timgl** [comment]
Can't we just replace this with a clickhouse query? Full text search over properties is pretty quick.

**Comment by neilkakkar** [comment]
Fixed via https://github.com/PostHog/posthog/pull/10868
